astropy/io/fits/hdu/hdulist.py

```diff
@@ -689,6 +698,13 @@ def writeto(self, fileobj, output_verify='exception', clobber=False,
             hdu._prewriteto(checksum=checksum)
             try:
                 hdu._writeto(hdulist._file)
+            except IOError, e:
+                free_space = get_free_space_in_dir(os.getcwd())
```

Comment:
This will get the free space in the current working directory, not necessarily the free space in the directory the file was written to.  You need to use something like `os.path.dirname(hdulist._file.name)`.  But be careful because for some files this might not actually correspond to a filesystem path (e.g. if `fileobj.file_like` is True), so also make sure the result of `dirname()` is a directory that actually exists (and probably move the `file_like` check to the top of the whole thing).